### PR TITLE
update timm commit

### LINKED ISSUE #100903: Update timm version used in timm benchmarking suite and torchbench
Currently more than [half a year old](https://github.com/huggingface/pytorch-image-models/commit/6635bc3f7d06c6a0d0481803b24d6ad0004b61ac) version is used for timm suite and [0.8.11.dev0](https://github.com/huggingface/pytorch-image-models/commit/45af496197a6e76aa6acd1332af11db65b63e97a) is used in torchbench. We should update to [0.8.21dev0](https://github.com/huggingface/pytorch-image-models/commit/5e64777804154b510ab41f0ace5ae30551bc991b) and keep torchbench and pt2 suites in sync. 
This version enabled using fast scaled dot product attention on vision transformer models, and we should benchmark it. 
Note: when updating timm suite, I expect that our speed-ups will meaningfully change (because of sdpa), we should make sure that absolute times improve, and provide a way to compare pre/post numbers. We should also make sure that fast path is actually taken in sdpa.

cc @ezyang @soumith @msaroufim @wconstab @bdhirsh @anijain2305 @xuzhao9 for torchbench

### PR DESCRIPTION
**Pull request by ngimel**
Fixes #100903

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: .ci/pytorch/test.sh [hunk 1]
```diff
@@ -986,9 +986,9 @@ elif [[ "${TEST_CONFIG}" == *torchbench* ]]; then
   if [[ "${TEST_CONFIG}" == *inductor_torchbench_smoketest_perf* ]]; then
     checkout_install_torchbench hf_Bert hf_Albert timm_vision_transformer
     PYTHONPATH=$(pwd)/torchbench test_inductor_torchbench_smoketest_perf
-  else
-    checkout_install_torchbench
-    PYTHONPATH=$(pwd)/torchbench test_dynamo_benchmark torchbench "$id"
+#  else
+#    checkout_install_torchbench
+#    PYTHONPATH=$(pwd)/torchbench test_dynamo_benchmark torchbench "$id"
```
Comment (ezyang) [inline .ci/pytorch/test.sh:991]: huh, how come?

**Review by ezyang — APPROVED** [review]
Oh, you're just disabling the other configs. Approving, don't forget to revert when you're done

**Comment by ngimel** [comment]
YEah that's to get faster perf numbers. Speedup is good, pass rate isn't.

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**